src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -2848,4 +2972,206 @@ public function is_development_environment() {
 		return in_array( wp_get_environment_type(), array( 'development', 'local' ), true );
 	}
 
+	/**
+	 * Returns a list of headers and its verification callback to verify if page cache is enabled or not.
+	 *
+	 * Note: key is header name and value could be callable function to verify header value.
+	 * Empty value mean existence of header detect page cache is enabled.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @return array List of client caching headers and their (optional) verification callbacks.
+	 */
+	public function get_page_cache_headers() {
+
+		$cache_hit_callback = static function ( $header_value ) {
+			return false !== strpos( strtolower( $header_value ), 'hit' );
+		};
+
+		$cache_headers = array(
+			'cache-control'          => static function ( $header_value ) {
+				return (bool) preg_match( '/max-age=[1-9]/', $header_value );
+			},
+			'expires'                => static function ( $header_value ) {
+				return strtotime( $header_value ) > time();
+			},
+			'age'                    => static function ( $header_value ) {
+				return is_numeric( $header_value ) && $header_value > 0;
+			},
+			'last-modified'          => '',
+			'etag'                   => '',
+			'x-cache'                => $cache_hit_callback,
+			'x-proxy-cache'          => $cache_hit_callback,
+			'cf-cache-status'        => $cache_hit_callback,
+			'x-kinsta-cache'         => $cache_hit_callback,
+			'x-aruba-cache'          => $cache_hit_callback,
+			'x-cache-enabled'        => static function ( $header_value ) {
+				return 'true' === strtolower( $header_value );
+			},
+			'x-cache-disabled'       => static function ( $header_value ) {
+				return ( 'on' !== strtolower( $header_value ) );
+			},
+			'cf-apo-via'             => static function ( $header_value ) {
+				return false !== strpos( strtolower( $header_value ), 'tcache' );
+			},
+			'x-srcache-store-status' => $cache_hit_callback,
+			'x-srcache-fetch-status' => $cache_hit_callback,
+			'cf-edge-cache'          => static function ( $header_value ) {
+				return false !== strpos( strtolower( $header_value ), 'cache' );
+			},
+		);
```

Comment:
Since there is a filter to extend the list of headers, should we perhaps not add SaaS-specific headers (such as the Cloudflare ones prefixed with `cf-*`) to core, but instead let them add this with their own plugin, which they tell people to use for WP sites any way?

That way, core should focus only on what nginx and apache would add, as the two webservers officially supported, and we don't need ot maintain a list of headers that may or may not be the same for the foreseeable future.

Reply:
that makes sense, though it also makes sense to have the most used ones already in place. I'm good with both.

Reply:
also tagging @felixarntz @spacedmonkey

Reply:
Yeah, I'm just vary of putting anything relating to 3rd parties into core, as we do not wish to show favoritism one way or the other as well (`x-kinsta-cache` and `x-aruba-cache` stand out immediately for example, as these are hosts, and would then have an "unfair advantage" so to speak).

Reply:
@manuelRod I agree with @Clorith on not including third party specific identifiers here. While I get the idea and I wish we could, it _does_ give them an unfair advantage which is something we have to avoid in WordPress core. Could you remove anything that is not "neutral" from the implementation here?

It would be great to make a mini plugin available, e.g. as a Gist, which adds all of these identifiers to the list via the filter. It could also be linked in a developer note later as an example for how to use the filter.

Reply:
LGTM!